aesd-char-driver/aesd-circular-buffer.c

```diff
@@ -34,47 +34,30 @@ struct aesd_buffer_entry *aesd_circular_buffer_find_entry_offset_for_fpos(struct
     */
     
 	int count = 0;
-	bool entry = false;
     	int position = buffer->out_offs;
-
-	if  (true == buffer->full)
-	{
-		count = AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED;
-	}	
-	else
-	{
-		if(buffer->out_offs < buffer->in_offs)
-		{
-			count = buffer->out_offs - buffer->in_offs;
-		}
-		else if ( buffer->out_offs > buffer->in_offs)
-		{
-			count = AESDCHAR_MAX_WRITE_OPERATIONS_SUPPORTED +1 + buffer->out_offs - buffer->in_offs ;
-		}
-		else{
-			return NULL;
-		}
-	}
-
-	while(count && !entry)
+	char_offset = char_offset+1;
```

Comment:
@akshithaluguri22 I'd recommend leaving the `char_offset` to be zero referenced and checking for `size > char_offset` for end condition.